tests/test_push_staged_cgw.py

```diff
@@ -0,0 +1,128 @@
+from pubtools._content_gateway.push_staged_cgw import PushStagedCGW, entry_point
+import pytest
+from tests.fake_cgw_client import TestClient
+import os
+from pushsource import CGWPushItem, FilePushItem
+import json
+from attrs import asdict
+
+try:
+    import mock
+except ImportError:
+    from unittest import mock
+
+test_dir = os.path.dirname(__file__)
+yaml_file_path = 'test_data/test_cgw_push_staged.yaml'
+invalid_push_item_path_yaml_file = 'test_data/test_invalid_cgw_push_staged.yaml'
+
+
+@pytest.fixture()
+def target_setting():
+    return {
+        "username": "foo",
+        "password": "bar",
+        "server_name": "http://fake/server/"
+    }
+
+
+class Struct:
+    def __init__(self, **entries):
+        self.__dict__.update(entries)
+
+
+def get_pulp_push_item():
+    pulp_push_item = {
+        "repository_memberships": "test",
+        "content_type_id": "test",
+        "cdn_published": "test",
+        "cdn_path": "test",
+        "display_order": "test",
+        "version": "v1",
+        "description": "test",
+        "sha256sum": "test",
+        "size": "test",
+        "path": "test"
+    }
+
+    return Struct(**pulp_push_item)
```

Comment:
you can use mock.MagicMock() instead of custom class

Reply:
I agree that I could have simply mocked this entire **pulp_push_item** but from the code reading perspective, one may not understand the value of **pulp_push_item**. So for a clear understanding, I am making a dictionary object which will be returned for pulp_push_item. 

Mocking **pulp_push_item** is also working fine here. What’s your suggestion? 
Should I mock it or keep it as it is?

Reply:
It can work also with custom class so you don't need to mock it. That was just suggestion to think about